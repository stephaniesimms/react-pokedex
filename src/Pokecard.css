.Pokecard {
    background-color: gray;
    border-radius: 5px;
    margin: 5px auto;
    max-width: 125px;
    position: relative;
    padding: 1px;
}

.Pokecard-name {
    color: blue;
    font-family: Arial, Helvetica, sans-serif;
    font-display: bold;
    text-align: center;
}

.Pokecard-type, 
.Pokecard-exp {
    font-family: Arial, Helvetica, sans-serif;
    font-size: 70%; 
    text-align: center;
    margin: 1px;
    padding-bottom: 4px;
}

.Pokecard-img {
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-top: -20px;
    
}