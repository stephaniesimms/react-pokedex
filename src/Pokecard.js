import React, { Component } from "react";
import "./Pokecard.css";

class Pokecard extends Component {
  render() {
    const { id, name, type, base_experience } = this.props;
    let imgUrl = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${id}.png`
    return (
      <div className="Pokecard">
        <h4 className="Pokecard-name">{name}</h4>
        <img className="Pokecard-img" src={imgUrl} alt={name}></img>
        <p className="Pokecard-type">Type: {type}</p>
        <p className="Pokecard-exp">EXP: {base_experience}</p>
      </div>
    )
  }
}

export default Pokecard;