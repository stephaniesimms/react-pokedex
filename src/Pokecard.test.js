import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Pokecard from "./Pokecard";

it("renders without crashing", function() {
  shallow(<Pokecard />);
});

it("matches snapshot", function() {
  let wrapper = shallow(<Pokecard />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});