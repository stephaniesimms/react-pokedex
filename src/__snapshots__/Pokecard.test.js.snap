// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Pokecard"
>
  <h4
    className="Pokecard-name"
  />
  <img
    className="Pokecard-img"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/undefined.png"
  />
  <p
    className="Pokecard-type"
  >
    Type: 
  </p>
  <p
    className="Pokecard-exp"
  >
    EXP: 
  </p>
</div>
`;
